WEBSITE-129 - Add pipeline steps reference documentation.

kwhetstone opened the pull request:
Static files for the pipeline-compatible steps.  They've been slightly tweaked so there's at least a better description for each of the steps.  These are all the plugins that I know are pipeline compatible at the moment.  More can be added as they're found.

The index page is a listing of all the plugins with their steps & a short description.  This way, the user will know what plugin they'll have to install.

rtyler commented:
@kwhetstone I'm confused, I thought these were going to be generated such that we could download them as part of the build process?

kwhetstone commented:
That's problematic at the moment (looks like connectivity issues? still trying to work that out) and I'm addressing the feedback on WEBSITE-129, so I went ahead and created the documents to be ready for the jenkins.io website launch.  I wanted to at least have some doc than no doc.

rtyler commented:
@kwhetstone so long as you promise to automate this, I'm fine with merging it.

If this doesn't get automated, then I will delete this from the tree. Stale documentation is IMO worse than no documentation
